intersect.d.ts

```diff
@@ -0,0 +1,91 @@
+
+/**
+ * Find or counts the intersections between two SVG paths.
+ *
+ * Returns a number in counting mode and a list of intersections otherwise.
+ *
+ * A single intersection entry contains the intersection coordinates (x, y)
+ * as well as additional information regarding the intersecting segments
+ * on each path (segment1, segment2) and the relative location of the
+ * intersection on these segments (t1, t2).
+ *
+ * The path may be an SVG path string or a list of path components
+ * such as `[ [ 'M', 0, 10 ], [ 'L', 20, 0 ] ]`.
+ *
+ * @example
+ *
+ * var intersections = findPathIntersections(
+ *   'M0,0L100,100',
+ *   [ [ 'M', 0, 100 ], [ 'L', 100, 0 ] ]
+ * );
+ *
+ * // intersections = [
+ * //   { x: 50, y: 50, segment1: 1, segment2: 1, t1: 0.5, t2: 0.5 }
+ * //
+ *
+ * @param {String|Array<PathDef>} path1
```

Comment:
Shouldn't `PathDef` be `PathComponent` (the exported type)?

Reply:
Yes. Converting TypeScript: If explicit type annotations are set, the type hints in the corresponding JSDoc are ignored. It wouldn't be an error, but it would be inconsistent.